crates/router/src/core/payments/tokenization.rs

```diff
@@ -208,9 +208,15 @@ where
                     .await?
                 };
 
-                let pm_card_details = resp.card.as_ref().map(|card| {
-                    PaymentMethodsData::Card(CardDetailsPaymentMethod::from(card.clone()))
-                });
+                #[cfg(feature = "v2")]
+                let pm_card_details = match &resp.payment_method_data {
+                    Some(api::PaymentMethodResponseData::Card(card_data)) => Some(
+                        PaymentMethodsData::Card(CardDetailsPaymentMethod::from(card_data.clone())),
+                    ),
+                    _ => None,
+                };
+                #[cfg(not(feature = "v2"))]
+                let pm_card_details = resp.card.clone().map(CardDetailsPaymentMethod::from);
```

Comment:
Use function based approach instead doing inline cfg's. Follow the same in other places too.